Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,143 @@
 use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
 use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
-use Smartbox\Integration\FrameworkBundle\Service;
 
-class AsyncQueueConsumer extends Service implements ConsumerInterface, LoggerAwareInterface
+/**
+ * Class PhpAmqpSignalConsumer.
+ */
+class AsyncQueueConsumer extends AbstractAsyncConsumer
 {
-    use IsStopableConsumer;
     use UsesSmartesbHelper;
     use UsesSerializer;
-    use LoggerAwareTrait;
     use UsesExceptionHandlerTrait;
 
     /**
-     * @var QueueManager
+     * Consumer identifier name.
      */
-    private $manager;
+    const CONSUMER_TAG = 'amqp-consumer-%s-%s';
 
     /**
      * @var string
      */
-    private $format;
+    private $format = QueueDriverInterface::FORMAT_JSON;
 
     /**
-     * AsyncQueueConsumer constructor.
+     * @var PhpAmqpLibDriver
+     */
+    private $driver;
+
+    /**
+     * Set the driver to this class with the properties fulfilled.
      *
-     * @param QueueManager $manager
-     * @param string       $format
+     * @param $driver
      */
-    public function __construct(QueueManager $manager, string $format = 'json')
+    public function setDriver($driver)
     {
-        parent::__construct();
-        $this->manager = $manager;
-        $this->format = $format;
+        $this->driver = $driver;
     }
 
     /**
-     * Consumes messages from the given $endpoint until either the expirationCount reaches 0 or ::stop() is called.
-     *
-     * @param EndpointInterface $endpoint
-     *
-     * @throws \Exception
-     *
-     * @return bool
+     * {@inheritdoc}
      */
-    public function consume(EndpointInterface $endpoint)
+    protected function initialize(EndpointInterface $endpoint)
     {
-        $handler = new AmqpQueueHandler($endpoint, (int) $this->expirationCount, $this->format, $this->serializer);
-        $handler->setExceptionHandler($this->getExceptionHandler());
-
-        if ($this->smartesbHelper) {
-            $handler->setSmartesbHelper($this->smartesbHelper);
-        }
-        if ($this->logger) {
-            $handler->setLogger($this->logger);
+        if (!$this->driver->isConnected()) {
+            $this->driver->connect();
         }
-
-        try {
-            $this->manager->getQueue($this->getQueueName($endpoint))->consume($handler);
-        } finally {
-            $this->manager->disconnect();
-        }
-
-        return true;
     }
 
     /**
+     * Returns the consumer name.
+     *
      * {@inheritdoc}
      */
     public function getName()
     {
-        return 'async_queue_consumer';
+        return sprintf(self::CONSUMER_TAG, gethostname(), getmypid());
```

Comment:
Is this method needed to be public?

Reply:
This method is inherit from the interface that is also public.

Reply:
Ok